site/cypress/e2e/cypress-feature/grading_stats.spec.js

```diff
@@ -6,155 +6,43 @@ describe('Test cases for grading stats', () => {
         });
         it(`${user} view should be accurate for teams.`, () => {
             cy.get('a[href*="/sample/gradeable/grading_team_homework/grading/details"]').click();
-
-            /*
-            if (user === 'ta' || user === 'grader') { // Close Grader Responsibilities popup.
-                cy.get('.popup-window .form-title .btn.btn-default.close-button.key_to_click').click();
-            }
-            */
-
             cy.get('a[href*="/sample/gradeable/grading_team_homework/grading/status"]').click();
-
+            cy.get('#filters').click();
+            cy.get('#bad_submissions').click();
+            cy.get('#apply_button').click();
 
             const text = cy.get('#left-grading-stats');
             text.should('contain', 'Students on a team: 101/101 (100%)');
             text.should('contain', 'Number of teams: 36');
-            text.should('contain', 'Teams who have submitted on time: 27 / 36 (75%)');
+            text.should('contain', 'Teams who have submitted: 27 / 36 (75%)');
             text.should('contain', 'Section 1: 2 / 4 (50.0%)');
         });
 
         it(`${user} view should be accurate for grades.`, () => {
             cy.get('a[href*="/sample/gradeable/grading_homework/grading/details"]').click();
-
-            /*
-            if (user === 'ta' || user === 'grader') { // Close Grader Responsibilities popup.
-                cy.get('.popup-window .form-title .btn.btn-default.close-button.key_to_click').click();
-            }
-            */
-
             cy.get('a[href*="/sample/gradeable/grading_homework/grading/status"]').click();
```

Comment:
Instead of clicking two buttons, you can just visit the final page, and it should be faster, and more efficient.